### PR TITLE
Fix: catastrophic backtracking and esbuild platform

### LINKED ISSUE #219: onMount not being called with esbuild
I experienced this problem whilst putting together https://github.com/Elderjs/elderjs/pull/217 but it always went away again if I did a clean install of @elderjs.

Now however, even if I clone the template as stated in the documentation with `npx degit Elderjs/template elderjs-app`, the `onMount` hook from `svelte` still isn't called when running the local server with `esbuild`. Resulting in the clock not ticking for example.

I have tried different node versions (12, 14, 17) and tried downgrading esbuild (even back to 0.12.4) but nothing seems to help.  

Using rollup with `yarn dev` works fine...

Do you have any idea what the problem might be? Or where I could carry on digging?

### LINKED ISSUE #235: Catastrophic backtracking bug
https://github.com/Elderjs/elderjs/blob/bf528f1a20da10586467ce6c891d5bf3eafe3c79/src/partialHydration/mountComponentsInHtml.ts#L17
This runs forever when:

1. There are many ejs components.
2. The last item `<\/\1>` doesn't match.

This happens naturally when there is another tag stick with ejs component. Here is an example:
```html
<p><div class="ejs-component" data-ejs-component="Fumen" data-ejs-props="{blabla}" data-ejs-options="{blabla}"></div>
... and more ejs components ...
```
1. The first capturing group matches `p><div`.
2. The last item `</p><div>` can't match `</div>` in the file.
3. All `[^]*?` expand down to the entire file.
4. Still doesn't match. So it backtracks and test the previous `[^]*?` down to the entire file...

I think it will be better to write it as:
```js
 /<([^<]+) class="ejs-component[^"]*" data-ejs-component="([A-Za-z]+)" data-ejs-props="({[^"]*})" data-ejs-options="({[^"]*})"><\/\1>/gim
```
Ensure that:

1. The last item is a valid tag.
2. Even if the last item doesn't match, it doesn't expand down to the entire file.

---

After the fix, the build time reduces from 5 minutes to 9 seconds.

### PR DESCRIPTION
**Pull request by eight04**
Fixes #235 
Fixes #219

### CODE REVIEW

File: src/esbuild/esbuildBundler.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ const svelteHandler = async ({ elderConfig, svelteConfig, replacements, restartH
       },
       format: 'esm',
       target: ['es2020'],
-      platform: 'node',
+      platform: 'browser',
```
Comment (nickreese) [inline src/esbuild/esbuildBundler.ts:158]: So obvious. Thank you for the good catch here.

File: src/partialHydration/__tests__/mountComponentsInHtml.spec.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ describe('#mountComponentsInHtml', () => {
 
     mountComponentsInHtml.default({
       page,
-      html: `<div class="svelte-datepicker"><div class="ejs-component" data-ejs-component="Datepicker" data-ejs-props="{ "a": "b" }" data-ejs-options="{ "loading": "lazy" }"></div></div>`,
+      html: `<div class="svelte-datepicker"><div class="ejs-component" data-ejs-component="Datepicker" data-ejs-props="{ &quot;a&quot;: &quot;b&quot; }" data-ejs-options="{ &quot;loading&quot;: &quot;lazy&quot; }"></div></div>`,
```
Comment (nickreese) [inline src/partialHydration/__tests__/mountComponentsInHtml.spec.ts:38]: This is indeed correct as the props and options come out of `JSON.stringify` and `escapeHtml` in `inlineSvelteComponent.ts` then are reversed during the mounting process. TY

**Review by nickreese — APPROVED** [review]
Thank you so much for the PR. 

Could I ask a test to cover the regex change? It is clear that you've got a deep understanding of the issue and your regex is better than mine. 👍 

Ready to go besides that. 

Thanks for all of the support on the project.

**Comment by eight04** [comment]
A performance test is added. You can observe the performance issue with following commands:
```sh
# run the performance test
npx jest src/partialHydration/__tests__/mountComponentsInHtml.spec.ts

# use old regex and run again. It will hang and eat up your CPU
git checkout master -- src/partialHydration/mountComponentsInHtml.ts
npx jest src/partialHydration/__tests__/mountComponentsInHtml.spec.ts
```

**Review by nickreese — APPROVED** [review]
Looks great. Thank you.

**Pull request merged**